src/test/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilderUnitTest.java

```diff
@@ -123,6 +122,26 @@ public void testSetOutputFile() throws IOException {
     }
 
     @Test
+    public void testDetermineOutputType() {
+        Assert.assertEquals(OutputType.VCF, VariantContextWriterBuilder.determineOutputTypeFromFile(this.vcf));
+        Assert.assertEquals(OutputType.BCF, VariantContextWriterBuilder.determineOutputTypeFromFile(this.bcf));
+        Assert.assertEquals(OutputType.VCF_STREAM, VariantContextWriterBuilder.determineOutputTypeFromFile(new File("/dev/stdout")));
+
+        // Test symlinking
+        try {
+            final Path link = Files.createTempFile("foo.", ".tmp");
```

Comment:
what happens if the symlink is purposefully confusing? I'm not sure I know what I want the answer to be, but I'd like the behaviour to be clear....so if `link.vcf` points to a `file.bcf` what happens?